src/main/java/redis/clients/jedis/exceptions/JedisBatchOperationException.java

```diff
@@ -0,0 +1,10 @@
+package redis.clients.jedis.exceptions;
+
+public class JedisBatchOperationException extends JedisException {
```

Comment:
Can you make it extend "JedisDataException" this way it won't make a backward issue?

Reply:
@gkorland It was done deliberately.

[According to marcosnils](https://github.com/xetorthio/jedis/issues/1183#issuecomment-166799828) @marcosnils 
> I thought about this and I believe the best approach would be to change the type of the exception to `JedisException` instead of `JedisDataException` and to improve the logging message. ... Currently `JedisDataException` is used mainly when Redis returns an error about the operation. The pipeline/transaction validation we're doing it's something from the Jedis side, so it doesn't make sense to return this type of exception. ...

[According to HeartSaVioR](https://github.com/xetorthio/jedis/issues/1183#issuecomment-169230449) @HeartSaVioR 
> ... I also think we shouldn't use JedisDataException which is not ERR from Redis side. ...

I agree with both of them. And that's why I crafted this PR.

Reply:
@gkorland Please check #2360

Reply:
i agree @gkorland advice, we should be compatible with `JedisDataException`, because maybe user code has `catch JedisDataException`.

So, the inheritance path is ： JedisBatchOperationException extends JedisDataException extends JedisException.

Reply:
@yangbodong22011 WDYT about #2360 ?